File: tutorials/clojure/03-hello-cli/test/hello/cli_test.clj
(ns hello.cli-test
  (:require [clojure.test :refer :all]
            [hello.cli :refer :all]))

(deftest parse-args-test
  (is (= {:name "world" :shout false} (parse-args [])))
  (is (= {:name "Alice" :shout false} (parse-args ["--name" "Alice"])))
  (is (= {:name "world" :shout true} (parse-args ["--shout"])))
  (is (= {:name "Bob" :shout true} (parse-args ["--name" "Bob" "--shout"]))))

(deftest format-greeting-test
  (is (= "Hello, world!" (format-greeting {:name "world" :shout false})))
  (is (= "Hello, Alice!" (format-greeting {:name "Alice" :shout false})))
  (is (= "HELLO, BOB!" (format-greeting {:name "Bob" :shout true}))))

File: tutorials/java/03-hello-cli/src/test/java/hello/CliTest.java
package hello;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class CliTest {

    @Test
    void defaultOptions() {
        var opts = Options.parse(new String[]{});
        assertEquals("World", opts.name());
        assertFalse(opts.shout());
    }

    @Test
    void parseName() {
        var opts = Options.parse(new String[]{"--name", "Alice"});
        assertEquals("Alice", opts.name());
    }

    @Test
    void parseShout() {
        var opts = Options.parse(new String[]{"--shout"});
        assertTrue(opts.shout());
    }

    @Test
    void greetDefault() {
        assertEquals("Hello, World!", Cli.run(new Options("World", false)));
    }

    @Test
    void greetWithShout() {
        assertEquals("HELLO, ALICE!", Cli.run(new Options("Alice", true)));
    }

    @Test
    void parseCombined() {
        var opts = Options.parse(new String[]{"--name", "Bob", "--shout"});
        assertEquals("Bob", opts.name());
        assertTrue(opts.shout());
    }
}
